tests/test_date.py

```diff
@@ -132,76 +205,180 @@ def test_periods(self, low, high, period, expected_results):
+
+    def given_dates(self, dates):
+        self.dates = dates
+
+    def given_period_low(self, low):
+        self.period_low = low
 
+    def given_invalid_period(self, invalid_period):
+        self.invalid_period = invalid_period
 
-class ParseDateWithFormats(unittest.TestCase):
+    def given_period_high(self, high):
+        self.period_high = high
 
-    def test_shouldnt_parse_invalid_date(self):
-        self.assertIsNone(date.parse_with_formats('yesterday', ['%Y-%m-%d'])['date_obj'])
+    def given_period(self, period):
+        self.period = period
+
+    def when_date_intersecting_generated(self):
+        self.intersected_dates = self.dates
+
+    def when_should_one_date_for_each_month(self):
+        self.intersected_dates = list(date.get_intersecting_periods(
+            self.period_low, self.period_high, period=self.period))
+
+    def when_empty_period_is_parsed(self):
+        self.intersected_dates = list(date.get_intersecting_periods(
+            self.period_low, self.period_high, period=self.period))
+
+    def then_should_one_date_for_each_month(self):
+        self.assertEquals(self.expected_results, self.intersected_dates)
+
+    def then_date_range_length_should_one(self):
+        self.assertEquals(1, len(self.intersected_dates))
+
+    def then_date_period_and_intersected_dates_should_equal(self):
+        self.assertEquals(self.period_low, self.intersected_dates[0])
+
+    def then_period_is_empty(self):
+        self.assertEquals([], self.intersected_dates)
+
+    def then_should_reject_easily_mistaken_dateutil_arguments(self):
+        with self.assertRaisesRegexp(ValueError, "Invalid period: {}".format(self.invalid_period)):
+            date.get_intersecting_periods(self.period_low, self.period, period=self.invalid_period).next()
+
+
+class ParseDateWithFormats(BaseTestCase):
+    def setUp(self):
+        super(ParseDateWithFormats, self).setUp()
+        self.dt_data = NotImplemented
+
+    def test_date_with_not_matching_format_is_not_parsed(self):
+        self.given_invalid_date('yesterday')
+        self.when_parse_invalid_date()
+        self.then_shouldnt_parse_invalid_date()
 
     def test_should_parse_date(self):
-        result = date.parse_with_formats('25-03-14', ['%d-%m-%y'])
-        self.assertEquals(datetime(2014, 3, 25).date(), result['date_obj'].date())
+        self.given_date_str('25-03-14')
+        self.given_date_format('%d-%m-%y')
+        self.given_date(datetime(2014, 3, 25))
+        self.when_should_parse_date()
+        self.then_should_parse_date()
 
     def test_should_use_current_year_for_dates_without_year(self):
-        today = datetime.today()
+        self.given_current_year()
+        self.when_should_use_current_year_for_dates_without_year()
+        self.then_should_use_current_year_for_dates_without_year()
+
+    @parameterized.expand([
+        param(given_date=datetime(2014, 8, 31),
+              mock_date=datetime(2014, 8, 12),
```

Comment:
Do you need a `mock_date` here? Can't you construct one from `given_date` parameter?
Also parameter name starting with `given_` is a bit confusing.